File: 628-maximum-product-of-three-numbers/maximum-product-of-three-numbers.cpp
class Solution {
public:
    int maximumProduct(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int n = nums.size();
        
        int p1 = nums[n - 1] * nums[n - 2] * nums[n - 3];
        int p2 = nums[0] * nums[1] * nums[n - 1];
        
        return max(p1, p2);
    }
};

/*
1. Agr sirf pos hai toh...last ke 3 
2. agr neg b hai toh... ya toh last ke 3 ya start k 2 and last
*/

File: 628-maximum-product-of-three-numbers/maximum-product-of-three-numbers.java
class Solution {
    public int maximumProduct(int[] A) {
        int n = A.length;

        Arrays.sort(A);
        
        long a = A[0];
        long b = A[1];
        long c = A[n - 1];
        long d = A[n - 2];
        long e = A[n - 3];

        return (int) Math.max(a * b * c, c * d * e);
    }
}